Simbody/src/GeneralForceSubsystem.cpp

```diff
@@ -34,12 +34,244 @@
 #include "simbody/internal/GeneralForceSubsystem.h"
 #include "simbody/internal/SimbodyMatterSubsystem.h"
 #include "simbody/internal/MultibodySystem.h"
+#include <iostream>
+#include <exception>
+#include <mutex>
+#include <thread>
 
 #include "ForceImpl.h"
 
+#include <memory>
+
+#define NUMNONPARALLELTHREADS 1
+#define NONPARALLELFORCESINDEX 0
 
 namespace SimTK {
 
+/** Calls calcForce() on all enabled forces. For an index of 0, this Task
+ calculates the force for all non-parallelized forces. For all other indices,
+ this Task executes a single parallelized force. */
+class CalcForcesTask : public ParallelExecutor::Task {
+public:
+
+    enum Mode {
+        All,
+        CachedAndNonCached,
+        NonCached
+    };
+
+    CalcForcesTask(ParallelExecutor& executor): m_executor(executor), m_mode(All) {}
```

Comment:
This leaves a bunch of uninitialized pointer data members. All data members should be initialized on construction. You shouldn't do that explicitly here though; instead replace the naked pointer data members with some kind of smart pointer that is self initializing. Possibly Simbody's `ReferencePtr<T>` might do what you need.